pallas-hardano/src/storage/immutable/mod.rs

```diff
@@ -59,13 +150,230 @@ pub fn read_blocks(dir: &Path) -> Result<impl Iterator<Item = FallibleBlock>, st
     Ok(iter)
 }
 
+/// Returns an iterator over the chain from the given point if the specific
+/// block is found, otherwise returns an error.
+///
+/// # Errors
+///
+/// * `Error::OriginMissing` - If the first block in the chain is not the
+///   genesis block.
+/// * `Error::CannotFindBlock` - If the specific block indicated by the `Point`
+///   value is not found.
+/// * `std::io::Error` - If an I/O error occurs.
+/// * `pallas_traverse::Error` - If the block cannot be decoded.
+///
+/// # Example
+///
+/// ```rust
+/// use std::path::Path;
+/// use std::error::Error;
+/// use crate::{Point, read_blocks_from_point};
+///
+/// fn main() -> Result<(), Box<dyn Error>> {
+///     let dir = Path::new("/path/to/blocks");
+///     let point = Point::Origin;
+///     
+///     match read_blocks_from_point(dir, point) {
+///         Ok(iterator) => {
+///             for block in iterator {
+///                 match block {
+///                     Ok(block_data) => {
+///                         println!("Block data: {:?}", block_data);
+///                     }
+///                     Err(error) => {
+///                         println!("Error reading block: {:?}", error);
+///                     }
+///                 }
+///             }
+///         }
+///         Err(error) => {
+///             println!("Error reading blocks from point: {:?}", error);
+///         }
+///     }
+///
+///     Ok(())
+/// }
+/// ```
+pub fn read_blocks_from_point(
+    dir: &Path,
+    point: Point,
+) -> Result<Box<dyn Iterator<Item = FallibleBlock>>, Box<dyn std::error::Error>> {
+    let names = build_stack_of_chunk_names(dir)?;
+
+    match point {
+        // Establish iterator from the beginning of the chain
+        Point::Origin => {
+            let mut iter = ChunkReaders(dir.to_owned(), names)
+                .map_while(Result::ok)
+                .flatten()
+                .peekable();
+
+            // check the first block
+            match iter.peek() {
+                Some(Ok(block_data)) => {
+                    let block = MultiEraBlock::decode(&block_data)?;
+                    // check that the first block is genesis
+                    if block.slot() == 0 && block.number() == 0 {
```

Comment:
Why do we need to check for `slot`?  since it is genesis

Reply:
For the genesis block `slot` equals to `0` same as for the `number`.
Generally it should be enough to check only for `number` field, but as this check does not cost too much for as and for additional validation I think it is better to check both these fields.